sale_margin_sync/models/stock_valuation_layer.py

```diff
@@ -1,12 +1,24 @@
 # Copyright 2019 Tecnativa - Carlos Dauden
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
-from odoo import models
+from odoo import api, models
 
 
 class StockValuationLayer(models.Model):
     _inherit = "stock.valuation.layer"
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        """Update purchase price from sale line linked to svl record
+        """
+        svls = super().create(vals_list)
+        if not self.env.context.get("skip_sale_margin_sync", False):
+            for svl in svls:
+                sale_line = svl.stock_move_id.sale_line_id
+                if sale_line:
+                    sale_line.purchase_price = svl.unit_cost
```

Comment:
why not something like?:

```suggestion
            svls.filtered("stock_move_id.sale_line_id").sale_margin_sync()
```